### ISSUE TITLE
RSH issue for 1D system

### ISSUE THREAD

**Issue by zww-4855** [pyscf/pyscf#623]
Hi, 

I am attempting to obtain band structure for a 1D system using range-separated hybrid(RSH) DFT functionals. I have attached my code at the bottom. The following runs successfully for normal Hartree-Fock, as well as LDA. The issue arises purely from RSH like HSE06 and CAM-B3LYP. 

```
import numpy as np
import matplotlib.pyplot as plt
import pyscf.pbc.gto as pbcgto
import pyscf.pbc.scf as pbcscf
import pyscf.pbc.dft as pbcdft
import sys,time


cell = pbcgto.Cell()
cell.build(unit = 'A',
           a = [[ 2.44923679,0,0],[0,30,0],[0,0,30]],
           mesh =[18, 105, 105],
           atom = '''
C 0.6351735810854515 15.339047075240025 0.0
H 0.6334106195406675 16.429045649535695 0.0
C 1.8140664189145486 14.660952924759975 0.0
H 1.8158293804593326 13.570954350464305 0.0
           ''',
           dimension=1,
           low_dim_ft_type= 'inf_vacuum',
           max_memory=120000,
           verbose = 6,
           rcut = 45,
           basis='ccpvdz')

# 1D kpt/kpath sampling
endpt=np.array([0.5,0.0,0.0])
npoints=15 #var
interval=endpt/(npoints-1)
band_kpts=[]
const=2*np.pi/2.44923679 #var
ranged=const/(npoints-1)
kpath=[]
kpath.append(0.0)
band_kpts.append([0.0,0.0,0.0])
totinter=0.0
a=0.0
for i in range(npoints-1):
    totinter=totinter+interval
    print('tot interval: ', totinter)
    a=a+ranged
    arr=totinter

    kpath.append(a)
    band_kpts.append(arr)

kpath=np.array(kpath)
print('before band kpts: ', band_kpts)
band_kpts=np.array(band_kpts)
band_kpts = cell.get_abs_kpts(band_kpts)#
print('kpath:', kpath)
print('band kpts:', band_kpts)
#
# band structure from Gamma point sampling
#
sp_points=np.array([kpath[0],kpath[-1]])
e_kn=[]
for kpt in band_kpts:
    kpts=cell.make_kpts([2,1,1]) + kpt
    mf = pbcdft.KRKS(cell,kpts).density_fit()
    mf.xc='cam-b3lyp'
    print(mf.kernel())
    e_kn.append(mf.mo_energy[0])    

# Normalize E

vbmax = -99
for en in e_kn:
    vb_k = en[cell.nelectron//2-1]
    if vb_k > vbmax:
        vbmax = vb_k
e_kn = [en - vbmax for en in e_kn]

#
# Plot band structure
#
au2ev = 27.21139
outx=open('x.dat','w')
outy=open('y.dat','w')
nbands = cell.nao_nr()
for n in range(nbands):
    outx.write(str(list(kpath)))
    outx.write(',')
    a=[]#[e[n]*au2ev for e in e_kn]
    for e in e_kn:
        a.append(e[n]*au2ev)
    outy.write(str(a))
    outy.write(',')
    print('*********')
    print("plotted kpath:", kpath)


```

The error message is as follows

```
    print(mf.kernel())
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/lib/misc.py", line 598, in aliased_fn
    return getattr(self, fname)(*args, **kwargs)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/scf/hf.py", line 1654, in scf
    conv_check=self.conv_check, **kwargs)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/scf/hf.py", line 144, in kernel
    vhf = mf.get_veff(mol, dm)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/dft/krks.py", line 103, in get_veff
    vklr = ks.get_k(cell, dm, hermi, kpts, kpts_band, omega=omega)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/scf/khf.py", line 554, in get_k
    with_j=False, omega=omega)[1]
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/scf/khf.py", line 563, in get_jk
    with_j, with_k, omega, exxdiv=self.exxdiv)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/df/df.py", line 703, in get_jk
    with_j, with_k, omega, exxdiv)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/df/aft.py", line 600, in _sub_df_jk_
    omega=None, exxdiv=exxdiv)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/df/df.py", line 719, in get_jk
    vk = df_jk.get_k_kpts(self, dm, hermi, kpts, kpts_band, exxdiv)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/df/df_jk.py", line 232, in get_k_kpts
    _ewald_exxdiv_for_G0(cell, kpts, dms, vk_kpts, kpts_band)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/df/df_jk.py", line 425, in _ewald_exxdiv_for_G0
    madelung = tools.pbc.madelung(cell, kpts)
  File "/Users/zacharywindom/Desktop/software/pyscf/f2/1_7_0/pyscf_qcintOPT/pyscf/pyscf/pbc/tools/pbc.py", line 446, in madelung
    return -np.einsum('i,i,i->', ZSI.conj(), ZSI, coulG*weights).real
ValueError: operands could not be broadcast together with shapes (396900,) (389376,)
```

Any help resolving this issue and understanding the origin of why HF and basic DFT work, while RSH DFT does not would be appreciated.

Thanks,
Zachary Windom

**Comment by fishjojo** [comment]
A workaround is to use even number of mesh points for the non-periodic axes, e.g., mesh = [18,106,106]

**Comment by fishjojo** [comment]
fixed in a7fc2661af09f6f6fba4e7f25431ac8686565c95
